oauthenticator/azuread.py

```diff
@@ -64,6 +65,16 @@ def get_tenant(self):
             app_log.info('ID4: {0}'.format(tenant_id))
             return tenant_id
 
+    def get_username_claim(self):
+        """
+        The claim to map to the jupyter username, such as `upn` or `unique_name`
+        See https://docs.microsoft.com/en-gb/azure/active-directory/develop/id-tokens
+        """
+        if hasattr(self, 'username_claim') and self.username_claim:
```

Comment:
I'd use the possibility to set a default value for the traitlet (https://traitlets.readthedocs.io/en/stable/using_traitlets.html#default-values-and-checking-type-and-value) to remove the need for testing for attributes etc. Probably means we can even remove all of `get_username_claim` and directly use `self.username_claim` further down

Reply:
Sounds good, please take a look at the updated version. Should we do the same for `get_tenant`?